### PR TITLE
Handle case where abstract overrides miss an implementation

### LINKED ISSUE #11170: AssertionError in rebindSuper
## Minimized code
The following compiles in a single package or removing protected modifier but fails in this form:
```Scala
package apackage {

  abstract class A {
    protected val x: Int
  }
  abstract class A2 {
    val x: Int
  }
}

package bpackage {
  import apackage._

  trait B extends A {
    println(x)
  }
  trait B2 extends A2 {
    println(x)
  }
}

package cpackage {
  import apackage._
  import bpackage._

  case class C(override protected val x: Int) extends A with B
  case class C2(override val x: Int) extends A2 with B2
}
```

## Output (click arrow to expand)
<details>

```scala
➜  snips ~/projects/dotty/bin/scalac -d /tmp i9329.scala
exception occurred while compiling i9329.scala
java.lang.AssertionError: assertion failed: cannot rebind method bpackage$B$$super$x, bpackage$B$$super$x x while compiling i9329.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: cannot rebind method bpackage$B$$super$x, bpackage$B$$super$x x
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at dotty.tools.dotc.transform.ResolveSuper$.rebindSuper(ResolveSuper.scala:115)
        at dotty.tools.dotc.transform.ResolveSuper.superAccessors$2$$anonfun$2(ResolveSuper.scala:52)
        at scala.collection.immutable.List.map(List.scala:246)
        at dotty.tools.dotc.transform.ResolveSuper.superAccessors$3(ResolveSuper.scala:53)
        at dotty.tools.dotc.transform.ResolveSuper.$anonfun$1(ResolveSuper.scala:55)
        at scala.collection.immutable.List.flatMap(List.scala:293)
        at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:55)
        at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:43)
        at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1004)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:363)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:256)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:427)
        at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:437)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:382)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:385)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:439)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:382)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:385)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:448)
        at dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:460)
        at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:300)
        at scala.collection.immutable.List.map(List.scala:246)
        at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:301)
        at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:185)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
        at dotty.tools.dotc.Run.runPhases$5(Run.scala:195)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:203)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
        at dotty.tools.dotc.Run.compileUnits(Run.scala:210)
        at dotty.tools.dotc.Run.compileSources(Run.scala:146)
        at dotty.tools.dotc.Run.compile(Run.scala:130)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
        at dotty.tools.dotc.Driver.process(Driver.scala:186)
        at dotty.tools.dotc.Driver.process(Driver.scala:155)
        at dotty.tools.dotc.Driver.process(Driver.scala:167)
        at dotty.tools.dotc.Driver.main(Driver.scala:194)
        at dotty.tools.dotc.Main.main(Main.scala)
```
</details>
Reported at scala/bug#12249

Split from #9329

### PR DESCRIPTION
**Pull request by odersky**
Fixes #11170

### CODE REVIEW

File: tests/neg/i11170a.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  trait B2 extends A2 {
+    println(x)
+  }
+}
+
+package cpackage {
+  import apackage._
+  import bpackage._
+
+  case class C(override protected val x: Int) extends A with B  // error
```
Comment (som-snytt) [inline tests/neg/i11170a.scala:26]: I think this should compile. "It's just a protected member." The ticket says it compiles if everything is in a single package.

Comment (odersky) [reply tests/neg/i11170a.scala:26]: But Scala 2.13 also refuses this one. Is that a known bug?

Comment (som-snytt) [reply tests/neg/i11170a.scala:26]: I see Scala 3 has a different definition of protected. It disalllows access to protected Java members from traits
```
illegal access to protected method test in class Base from trait Ext
```
at https://github.com/scala/bug/issues/3564

Maybe Scala just makes everything public if it wants to do something like that.

Comment (som-snytt) [reply tests/neg/i11170a.scala:26]: The Scala 3 ticket was noticed at https://github.com/scala/bug/issues/12249 which is just Scala classes. The other link is a Java base class with protected member. I haven't reflected deeply yet.

Comment (odersky) [reply tests/neg/i11170a.scala:26]: I am not sure about it either. But at least we don't crash anymore, and do the same thing as Scala 2.13 does.

**Review by bishabosha — APPROVED** [review]

**Pull request merged**